posix.mak

```diff
@@ -578,13 +573,19 @@ dpl-docs: ${DUB} ${STABLE_DMD}
 	DFLAGS="$(DPL_DOCS_DFLAGS)" ${DUB} build --root=${DPL_DOCS_PATH} \
 		--compiler=${STABLE_DMD}
 
-${STABLE_DMD}:
-	mkdir -p ${STABLE_DMD_ROOT}
-	TMPFILE=$$(mktemp deleteme.XXXXXXXX) && curl -fsSL ${STABLE_DMD_URL} > ${TMP}/$${TMPFILE}.zip && \
-		unzip -qd ${STABLE_DMD_ROOT} ${TMP}/$${TMPFILE}.zip && rm ${TMP}/$${TMPFILE}.zip
+# .tar.xz is archives are smaller (and don't need a temporary dir) -> prefer if available
+HAS_XZ := $(shell command -v xz 2> /dev/null)
+${STABLE_DMD_ROOT}/.downloaded:
+	@mkdir -p $(dir $@)
+ifdef HAS_XZ
+	curl -fSL --retry 3 $(subst .zip,.tar.xz,$(STABLE_DMD_URL)) | tar -Jxf - -C $(dir $@)
```

Comment:
The `tar.xz` balls has about half the size of the zip archive and apart from making the CI generate less traffic on the AWS bucket, it's quite noticeably faster on slower connections.
If there's consensus about this, I would propose to do the same for the DMD bootstrapping.